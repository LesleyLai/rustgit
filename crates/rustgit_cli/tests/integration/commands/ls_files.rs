use crate::common::{git, populate_folder, rustgit};
use lazy_static::lazy_static;
use std::{fs, path::PathBuf};
use test_utils::TEST_DIR;

lazy_static! {
    static ref WORKING_DIR: PathBuf = {
        let working_dir = TEST_DIR.join("ls-files");
        fs::create_dir(&working_dir).unwrap();

        git(&working_dir).init();

        assert!(rustgit(&working_dir).ls_files().is_empty());

        populate_folder(&working_dir);

        assert!(rustgit(&working_dir).ls_files().is_empty());

        git(&working_dir).stage(&["."]);

        working_dir
    };
}

#[test]
fn files() {
    let working_dir = &WORKING_DIR;
    insta::assert_snapshot!(rustgit(&working_dir).ls_files());
}

#[test]
fn stage() {
    let working_dir = &WORKING_DIR;
    insta::assert_snapshot!(rustgit(&working_dir).ls_files_stage());
}
