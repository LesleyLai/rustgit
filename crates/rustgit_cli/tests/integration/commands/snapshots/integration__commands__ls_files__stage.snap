---
source: crates/rustgit_cli/tests/integration/commands/ls_files.rs
assertion_line: 37
expression: git(&working_dir).ls_files_stage()
---
100644 26abb69f3d99e9fff02c20c26cc0a37dcb8dac03 0	dir1/file_in_dir1_1
100644 5a295e4de25bdc0c780038c7f422f10bc9d576e2 0	dir1/file_in_dir1_2
100644 ba33ff4884c58070963be49d0dc801daa333040a 0	dir2/file_in_dir2_1
100644 b6fc4c620b67d95f953a5c1c1230aaab5db5a1b0 0	file1.txt
